registry.go

```diff
@@ -104,6 +105,11 @@ type Layer interface {
 
 	// CreatedAt returns the time this layer was created.
 	CreatedAt() time.Time
+
+	// ServeHTTP allows a layer to serve itself, whether by providing
+	// a redirect directly to the content, or by serving the content
+	// itself
+	ServeHTTP(w http.ResponseWriter, r *http.Request)
```

Comment:
I am slightly conflicted here: a blob should be able to indicate that it cannot serve itself. It might be better to have a `Handler() (http.Handler, error)` method on `Layer`. Only call `(http.Handler).ServeHTTP` when this method returns without an error.